seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java

```diff
@@ -73,12 +81,52 @@ public class SeaTunnelServer implements ManagedService, MembershipAwareService,
 
     private final SeaTunnelConfig seaTunnelConfig;
 
+    /**
+     * IMap key is jobId and value is a Tuple2
+     * Tuple2 key is JobMaster init timestamp and value is the jobImmutableInformation which is sent by client when submit job
+     * <p>
+     * This IMap is used to recovery runningJobInfoIMap in JobMaster when a new master node active
+     */
+    private IMap<Long, RunningJobInfo> runningJobInfoIMap;
+
+    /**
+     * IMap key is one of jobId {@link org.apache.seatunnel.engine.server.dag.physical.PipelineLocation} and
+     * {@link org.apache.seatunnel.engine.server.execution.TaskGroupLocation}
+     * <p>
+     * The value of IMap is one of {@link JobStatus} {@link org.apache.seatunnel.engine.core.job.PipelineState}
+     * {@link org.apache.seatunnel.engine.server.execution.ExecutionState}
+     * <p>
+     * This IMap is used to recovery runningJobStateIMap in JobMaster when a new master node active
+     */
+    IMap<Object, Object> runningJobStateIMap;
```

Comment:
It's strange jobID map PipelineState, does JobStatus can't contains PipelineState?

Reply:
> It's strange jobID map PipelineState, does JobStatus can't contains PipelineState?

The key-value may be `jobId-JobStatus`, `PipelineLocaltion-PipelineState`, `TaskGroupLocation-ExecutionState`.

Reply:
It's hard to read, use one map put different key value type

Reply:
> It's hard to read, use one map put different key value type

Every IMap will create many storage file in hdfs, So I think we  should try to reduce the number of IMap.

Reply:
Fine, but  I think this IMAP needs to be refactored in the future

Reply:
> Fine, but I think this IMAP needs to be refactored in the future

Ok